code/datums/antags/traitor.dm

```diff
@@ -0,0 +1,231 @@
+/datum/antag/traitor
+	name = "Traitor"
+	antagtag = "traitor"
+
+	var/obj/item/device/uplink/uplink
+	var/starting_telecrystals = 20
+	var/given_codewords = FALSE
+
+/datum/antag/traitor/proc/inform_of_codewords(datum/mind/user)
+	var/mob/mob = user.current
+	mob << "<U><B>The Syndicate provided you with the following \
+		information on how to identify their agents:</B></U>"
+	mob << "<B>Code Phrase</B>: <span class='danger'>\
+		[syndicate_code_phrase]</span>"
+	mob << "<B>Code Response</B>: <span class='danger'>\
+		[syndicate_code_response]</span>"
+
+	user.store_memory("<b>Code Phrase</b>: \
+		[syndicate_code_phrase]")
+	user.store_memory("<b>Code Response</b>: \
+		[syndicate_code_response]")
+
+	mob << "Use the code words in the order provided, during \
+		regular conversation, to identify other agents. Proceed with \
+		caution, however, as everyone is a potential foe."
+
+/datum/antag/traitor/equip(mob/living/carbon/human/mob)
+	if(!ishuman(mob))
+		mob << "<span class='warning'>The Syndicate do not provide equipment \
+			for non-human agents.</span>"
+
+	var/list/uplink_order = list("pda", "radio", "implant")
+	var/randomise = FALSE
+	if(randomise)
+		uplink_order = shuffle(uplink_order)
+
+	var/success = FALSE
+	while(uplink_order.len && !success)
+		var/item = popleft(uplink_order)
+		switch(item)
+			if("pda")
+				var/P = locate(/obj/item/device/pda) in mob.contents
+				success = equip_pda(mob, P)
+			if("radio")
+				var/R = locate(/obj/item/device/radio) in mob.contents
+				success = equip_radio(mob, R)
+			if("implant")
+				success = equip_implant(mob)
```

Comment:
Since traitor assignment isn't always during roundstart, we can't guarantee that they'll even have a radio. Also if we were feeling random this could be randomly picked. If people are concerned about balance, we can take like 4TC off the balance to "pay" for the uplink.